src/groupByKey.ts

```diff
@@ -2,6 +2,25 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const mainObj: GroupsMap<T> = {};
+
+  for (let i: number = 0; i < items.length; i++) {
+    const parameter = items[i][key];
+
+    if (typeof parameter === 'string' || typeof parameter === 'number') {
+      const keySort = parameter.toString();
+
+      if (!mainObj[keySort]) {
+        mainObj[keySort] = [];
+      }
```

Comment:
Instead of manually checking if the key exists and then initializing it with an empty array, you could use the logical OR assignment operator to streamline the code. For example: `mainObj[keySort] ||= [];`.